src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1007,6 +1108,7 @@ namespace xclcpuemhal2 {
         if (auto sec = xrt_core::xclbin::get_axlf_section(top, EMULATION_DATA)) {
           emuData = xclbininmemory + sec->m_sectionOffset;
           emuDataSize = sec->m_sectionSize;
+          getCuRangeIdx();
```

Comment:
Since we are in end phase of release , Can we take it up in next release. We were only concentrating on adding autorestart support for edge platforms.

Reply:
ok Ramya. please refactor this code in next PR.